samcli/local/docker/container.py

```diff
@@ -430,10 +446,27 @@ def _write_container_output(output_itr, stdout=None, stderr=None):
             # Iterator returns a tuple of (stdout, stderr)
             for stdout_data, stderr_data in output_itr:
                 if stdout_data and stdout:
-                    stdout.write(stdout_data)
+                    if isinstance(stdout, StreamWriter):
+                        stdout.write_bytes(stdout_data)
+                        stdout.flush()
+
+                    if isinstance(stdout, io.BytesIO):
+                        stdout.write(stdout_data)
+
+                    if isinstance(stdout, io.TextIOWrapper):
+                        stdout.buffer.write(stdout_data)
 
                 if stderr_data and stderr:
-                    stderr.write(stderr_data)
+                    if isinstance(stderr, StreamWriter):
+                        stderr.write_bytes(stderr_data)
+                        stderr.flush()
+
+                    if isinstance(stderr, io.BytesIO):
+                        stderr.write(stderr_data)
+
+                    if isinstance(stderr, io.TextIOWrapper):
+                        stderr.buffer.write(stderr_data)
```

Comment:
Worth it to move this to a helper method since the same logic happens with `stdout`?

Reply:
done